cvprac/cvp_api.py

```diff
@@ -3204,6 +3210,449 @@ def workspace_build_status(self, workspace_id, build_id):
+                    root (series)
+                    |- left-leafs (series)
+                       |- leaf1 (series)
+                       |    |- pre-mlag-check-l1
+                       |    |- upgrade1
+                       |    |- post-mlag-check-l1
+                       |- leaf2 (series)
+                            |- pre-mlag-check-l1
+                            |- upgrade1
+                            |- post-mlag-check-l1
+
+            Returns:
+                response (dict): A dict that contains...
+                Ex: {'value': {'key': {'id':cc_id,
+                      'time': '...'}
+        '''
+        payload = custom_cc
+        url = '/api/resources/changecontrol/v1/ChangeControlConfig'
+        # For on-prem check the version as it is only supported from 2021.2.0+
+        if not self.clnt.is_cvaas:
+            if self.clnt.apiversion is None:
+                self.get_cvp_info()
+            if self.clnt.apiversion >= 6.0:
+                self.log.debug('v6 ' + str(url) + ' ' + str(payload))
+                response = self.clnt.post(url, data=payload)
+                return response
+            elif self.clnt.apiversion < 6.0:
+                    self.log.warning('change_control_create_with_custom_stages: Change Control Resource APIs are supported'
+                                     ' from 2021.2.0 or newer.')
+                    return None
+        else:
+            return self.clnt.post(url, data=payload)
+
+    def change_control_create_for_tasks(self, cc_id, name, tasks, series=True):
+        ''' Create a simple Change Control for tasks using Resource APIs.
+            Supported versions: CVP 2021.2.0 or newer and CVaaS.
+
+            This function will create a change with either all Task actions in series or parallel. For custom
+            stage hierarchy the change_control_create_with_custom_stages() should be used.
+            Args:
+                cc_id (string): The ID for the new change control.
+                name (string): The name for the new change control.
+                tasks (list): A list of Task IDs as strings
+                    Ex: ['10', '11', '12']
+                series (bool): A flag for running tasks in series or
+                    in parallel. Defaults to True for running in series.
+            Returns:
+                response (dict): A dict that contains...
+                Ex: {'value': {'key': {'id':cc_id,
+                      'time': '...'}
+        '''
+        stages = {'values': {'root': {'name': 'root', 'rows': {'values': []}}}}
+        if series == True:
+            for index, task in enumerate(tasks):
+                stage_id = 'stage%d' % index
+                stages['values']['root']['rows']['values'].append({'values':[stage_id]})
+                stages['values'][stage_id] = {'action': {'args': {'values': {'TaskID': task}},'name':'task','timeout':3000},'name':stage_id}
```

Comment:
Wondering if this timeout of 3000 should be configurable?

Reply:
Comment from Tamas:

"UI seems to use 3000 always". Will leave this as is for now and make it configurable in future if necessary.